Data range in .vtkjs file

devangcx opened the pull request:
This PR essentially adds data-range to the .vtkjs file so that the front-end can use it. A new object named "data_ranges" is added in the index.json. This object has data-range for all the simulation data added to a Dataset (Say "Grid"). 

![data-ranges](https://user-images.githubusercontent.com/20086924/129615980-fdf173d6-2ad6-4c5e-ac02-e09b12418380.PNG)


If min and max values are provided in the config file, They will find their way here in this data_ranges object. In the image above, the range for "Daylight-Factor" comes from the values set by a user in the config file. Otherwise, an auto-calculated range based on the simulation data will be reported. In the image above, the range for "UDI" is auto-calculated by honeybee-vtk.

devangcx committed:
feat(config): use legend min and max as data range

devangcx committed:
fix(legend_parameter): fix type checking issues

devangcx committed:
refactor(schema): add a new object to capture data ranges

devangcx committed:
feat(types): collect data rages for all the fields

devangcx committed:
test(*): update and add unit tests

devangcx committed:
chore(valid.json): update file

mostaphaRoudsari reviewed (commented):
Hi @devngc, I find the term `data_range` a bit confusing knowing that it can also be a user input range which is not really data range. I see in your code you are using `legend_range` internally. I like that more.

I believe we can rely on the viewer and vtk.js library for calculating the range of data if not provided by user. Doing the check here in honeybee-vtk can end up being time-consuming for larger datasets.

devngc commented:
> I believe we can rely on the viewer and vtk.js library for calculating the range of data if not provided by user. Doing the check here in honeybee-vtk can end up being time-consuming for larger datasets.

@mostaphaRoudsari, this is being done [here](https://github.com/ladybug-tools/honeybee-vtk/blob/master/honeybee_vtk/types.py#L178). This auto-calculated range is being used by .vtkjs for Praview Glance and export-images command as well. Are you sure you want me to not report auto-calculated range?

mostaphaRoudsari reviewed (approved):
Hi @devngc! I see. You're basically using the same method that will be used by vtkjs. Makes sense!

This branch has conflicts. Feel free to merge it once you resolve these conflicts.

devangcx committed:
feat(schema): add a 'legend_range' object

devangcx committed:
refactor(config): add _get_legend_range function

This function helps in extracting min and max values of legend from the config data and applies these values as data_range to the data being mounted on a honeybee-vtk model.

devangcx committed:
refactor(types): add _get_data_range helper function

This function processes the data_range coming from the config file.

devangcx committed:
refactor(config): move legend_range validation to type module

Now all the legend_range and data_range validation happens in _get_data_range function in the types module.

devangcx committed:
refactor(types): bring legend_range validation here

Now all the legend_range and data_range validation happens in _get_data_range function in the types module.

devangcx committed:
test(schema_test): start unit tests for schema

devangcx committed:
test(*): add unit tests
